Server crashes on jobs with zero steps

Not if we should reject them right away, or run them and just call them a success.
No strong opinions either way, but at the moment we crash the server:
```
panic: runtime error: index out of range [0] with length 0

goroutine 116 [running]:
github.com/facebookincubator/contest/pkg/runner.(*TestRunner).run(0xc000737720, 0x99d4f8, 0xc000737c20, 0xc0008b4b00, 0xc000720180, 0x3, 0x3, 0x740, 0x1, 0x0, ...)
	github.com/facebookincubator/contest/pkg/runner/test_runner.go:257 +0x1d16
github.com/facebookincubator/contest/pkg/runner.(*TestRunner).Run(0xc000737720, 0x99d4f8, 0xc000737810, 0xc0008b4b00, 0xc000720180, 0x3, 0x3, 0x740, 0x1, 0x0, ...)
	github.com/facebookincubator/contest/pkg/runner/test_runner.go:133 +0x486
github.com/facebookincubator/contest/pkg/runner.(*JobRunner).Run(0xc000722690, 0x99d4f8, 0xc0000caeb0, 0xc000a1e140, 0x0, 0x0, 0x0, 0x0)
	github.com/facebookincubator/contest/pkg/runner/job_runner.go:232 +0x11d3
github.com/facebookincubator/contest/pkg/jobmanager.(*JobManager).runJob(0xc000139720, 0x99d4f8, 0xc0000ca2d0, 0xc000a1e140, 0x0)
	github.com/facebookincubator/contest/pkg/jobmanager/start.go:98 +0x2b3
created by github.com/facebookincubator/contest/pkg/jobmanager.(*JobManager).startJob
	github.com/facebookincubator/contest/pkg/jobmanager/start.go:80 +0x245
```

--- a/pkg/jobmanager/bundles.go
+++ b/pkg/jobmanager/bundles.go
@@ -75,6 +75,10 @@ func newBundlesFromSteps(ctx xcontext.Context, descriptors []*test.TestStepDescr
 		stepBundles = append(stepBundles, *tsb)
 	}
 
+	if len(stepBundles) == 0 {
+		return nil, fmt.Errorf("at least one test step is required per test")
+	}
+
 	return stepBundles, nil
 
 }

--- a/pkg/jobmanager/job.go
+++ b/pkg/jobmanager/job.go
@@ -35,6 +35,9 @@ func newJob(ctx xcontext.Context, registry *pluginregistry.PluginRegistry, jobDe
 	if err := jobDescriptor.Validate(); err != nil {
 		return nil, fmt.Errorf("could not validate job descriptor: %w", err)
 	}
+	if len(jobDescriptor.TestDescriptors) == 0 {
+		return nil, fmt.Errorf("cannot create job with zero tests")
+	}
 
 	runReportersBundle, finalReportersBundle, err := newReportingBundles(registry, jobDescriptor)
 	if err != nil {

--- a/pkg/jobmanager/job_test.go
+++ b/pkg/jobmanager/job_test.go
@@ -89,3 +89,70 @@ func TestDisabledTestDescriptor(t *testing.T) {
 	require.Len(t, result.Tests, 1)
 	require.Equal(t, result.Tests[0].Name, "TestEnabled")
 }
+
+func TestNewJobNoTests(t *testing.T) {
+	pr := pluginregistry.NewPluginRegistry(xcontext.Background())
+	// require.NoError(t, pr.RegisterTestStep(echo.Load()))
+	require.NoError(t, pr.RegisterTargetManager(targetlist.Load()))
+	require.NoError(t, pr.RegisterTestFetcher(literal.Load()))
+	require.NoError(t, pr.RegisterReporter(noop.Load()))
+
+	testDescriptors := []*test.TestDescriptor{}
+	jd := job.Descriptor{
+		TestDescriptors: testDescriptors,
+		JobName:         "Test",
+		Reporting: job.Reporting{
+			RunReporters: []job.ReporterConfig{
+				{Name: "noop"},
+			},
+		},
+	}
+
+	_, err := NewJobFromDescriptor(xcontext.Background(), pr, &jd)
+	require.Error(t, err)
+}
+
+func TestNewJobNoTestSteps(t *testing.T) {
+	pr := pluginregistry.NewPluginRegistry(xcontext.Background())
+	// require.NoError(t, pr.RegisterTestStep(echo.Load()))
+	require.NoError(t, pr.RegisterTargetManager(targetlist.Load()))
+	require.NoError(t, pr.RegisterTestFetcher(literal.Load()))
+	require.NoError(t, pr.RegisterReporter(noop.Load()))
+
+	testFetcherParams := `{
+	    "TestName": "TestDisabled",
+		"Steps": [
+		]
+	}`
+
+	targetManagerAcquireParameters := `{
+		"Targets": [
+			{
+				"ID": "id1",
+				"FQDN": "some-host.example.com"
+			}
+		]
+	}`
+
+	testDescriptors := []*test.TestDescriptor{
+		{
+			TargetManagerName:              "targetList",
+			TargetManagerAcquireParameters: []byte(targetManagerAcquireParameters),
+			TargetManagerReleaseParameters: []byte("{}"),
+			TestFetcherName:                "literal",
+			TestFetcherFetchParameters:     []byte(testFetcherParams),
+		},
+	}
+	jd := job.Descriptor{
+		TestDescriptors: testDescriptors,
+		JobName:         "Test",
+		Reporting: job.Reporting{
+			RunReporters: []job.ReporterConfig{
+				{Name: "noop"},
+			},
+		},
+	}
+
+	_, err := NewJobFromDescriptor(xcontext.Background(), pr, &jd)
+	require.Error(t, err)
+}

--- a/pkg/storage/limits/limits_test.go
+++ b/pkg/storage/limits/limits_test.go
@@ -26,6 +26,7 @@ import (
 	"github.com/linuxboot/contest/plugins/reporters/noop"
 	"github.com/linuxboot/contest/plugins/targetmanagers/targetlist"
 	"github.com/linuxboot/contest/plugins/testfetchers/literal"
+	"github.com/linuxboot/contest/plugins/teststeps/echo"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
@@ -91,9 +92,18 @@ func TestTestName(t *testing.T) {
 	require.NoError(t, err)
 	err = pluginRegistry.RegisterReporter(noop.Load())
 	require.NoError(t, err)
+	err = pluginRegistry.RegisterTestStep(echo.Load())
+	require.NoError(t, err)
 
 	testFetchParams, err := json.Marshal(&literal.FetchParameters{
 		TestName: strings.Repeat("A", limits.MaxTestNameLen+1),
+		Steps: []*test.TestStepDescriptor{{
+			Label: "abc",
+			Name:  "echo",
+			Parameters: test.TestStepParameters{
+				"text": []test.Param{*test.NewParam("\"abc\"")},
+			},
+		}},
 	})
 	require.NoError(t, err)
 

